### DISCUSSION TITLE
Image not rendering after sell-nft function

### DISCUSSION THREAD

**Discussion by NickBot09** [smartcontractkit/full-blockchain-solidity-course-py#1783] [Q&A]
I have built the marketplace using graph everything works but only the sell-nft feature is giving some problem. When approving it is asking for price in dollars as you can see below 
![image](https://user-images.githubusercontent.com/94431096/202473520-8b539f2f-8150-4f07-81af-43ee1e61e292.png)

But even after a successful transaction the nft doesn't render 
![image](https://user-images.githubusercontent.com/94431096/202474245-fa5181a7-5e34-4216-9db8-6480c50b2fda.png)

I have uploaded some code here for your reference https://github.com/NickBot09/nft-marketplace 
Any help would be appreciated

**Comment by cromewar** [comment]
Hello @NickBot09 
Have you waited enough time for the nft to get processed, this actions are asynchronous and might take some time.
you can share part of your code for extra support.

**Reply by NickBot09** [reply]
Yes I waited for like more than 15mins and I have shared my GitHub repo where I have uploaded my code you can take a look, only the sell-nft doens't work rest everything works perfectly

**Reply by cromewar** [reply]
@PatrickAlphaC a hand on this please, I'm struggling to find the way to go with this.

**Comment by NickBot09** [comment]
Yep @PatrickAlphaC please help

**Reply by PatrickAlphaC** [reply]
Looks like there is a lot to triage here... 

You can check a few things to make sure:

1. Is your smart contract 100% confirmed listed on-chain?
2. If you're using the graph, is the graph API picking up the event? 
3. Is your front end pulling all the data down?
